### ISSUE TITLE
no segments on my own data

### ISSUE THREAD

**Issue by 6NeverGiveup** [luigifreda/plvs#39]
Greetings! Thanks for your great job plvs!
I want to achieve segmentation function on my own data, but I find there's some problem i cant figure it out.
and my debug step as follow:
**①I can successfully obtained segmentation results on TUM data, use 'rgbd_tum';**
![plvs_tum](https://github.com/user-attachments/assets/863ca688-6232-4e4b-8533-2021ada8bc8a)
**②I modified some basic parameters (Camera calibration, and resolution is 1920*1080)in '/Monocular/TUM1.yaml', and i can successfully run monocular function on my own data, use 'mono_tum';**
![plvs_tum_mono](https://github.com/user-attachments/assets/15578097-face-4a8f-a4b0-ab92081f2766)
**③Im trying to use 'rgbd_tum' and modified parameters in 'RGB-D/TUM1.yaml', i can run it but no information displayed**
![plvs_data](https://github.com/user-attachments/assets/d6d4c344-a1fa-40d8-aa02-df1cfbb4bbcb)
and i enlarge the point cloud window, the key frame points cloud appear strange
![plvs_data_cloud](https://github.com/user-attachments/assets/af477002-6a57-4a41-b274-c989246f1b09)

I tried to work it out and catch some information, then i probably think it caused by depth. but i feel mistery. 
i dont know if the problem may caused by:
1. some modified parameters  in 'RGB-D/TUM1.yaml' file is wrong;
2. my depth image;
3. other

Thanks for your help！

**Comment by coutrol** [comment]
hello,bro,can give me some suggestion about my problem for ./build.sh?Thanks for your help

**Comment by luigifreda** [comment]
Hi,
Did you check your depth scale factor is correct?

**Comment by 6NeverGiveup** [comment]
> hello,bro,can give me some suggestion about my problem for ./build.sh?Thanks for your help

yes, what are the details of the problem?
I compiled and passed project by ./build.sh successfully, I guess you need to match libs version

**Comment by 6NeverGiveup** [comment]
> Hi, Did you check your depth scale factor `DepthMapFactor` is correctly set?

Thank u for reply！
I use default value 1000;
I didn't have a depth camera, my image data comes from rendering the model from different perspectives.
So initially, I didn't have depth image, I calculated the depth data for each frame of image data。
I use the distance from pixels to the camera imaging plane and project it to 0-255 using the maximum and minimum distance method.
could u please give me some debug advice?

**Comment by luigifreda** [comment]
> Thank u for reply！
> I use default value 5000;
> I didn't have a depth camera, my image data comes from rendering the model from different perspectives.
> So initially, I didn't have depth image, I calculated the depth data for each frame of image data。
> I use the distance from pixels to the camera imaging plane and project it to 0-255 using the maximum and minimum distance method.
> could u please give me some debug advice?

You are computing and passing a depth image in meters. In such a case, DepthMapFactor=1 should work fine. 
LMK.

**Comment by 6NeverGiveup** [comment]
> > > 您好，兄弟，可以给我一些关于我的问题的建议吗./build.sh？感谢您的帮助
> > 
> > 
> > 是的，问题的详细信息是什么？我成功编译并通过 ./build.sh 传递项目，我想您需要匹配 libs 版本
> 
> hi，can i get your qq for help？my number is 1274261172，Much appreciated。

sure，I added you, check it out

**Comment by 6NeverGiveup** [comment]
> @6NeverGiveup
> 
> > Thank u for reply！
> > I use default value 5000;
> > I didn't have a depth camera, my image data comes from rendering the model from different perspectives.
> > So initially, I didn't have depth image, I calculated the depth data for each frame of image data。
> > I use the distance from pixels to the camera imaging plane and project it to 0-255 using the maximum and minimum distance method.
> > could u please give me some debug advice?
> 
> AFAIU you are computing and passing a depth image in meters. In such a case, DepthMapFactor=1 should work fine. LMK.

Thanks for ur advice, but DepthMapFactor=1 still not show segments
and i tried some other value, while DepthMapFactor=40, there something show but seems wrong:
①There are too many layers in the RGB point cloud, causing distortion
![1wrongvslam](https://github.com/user-attachments/assets/4e256a85-1524-4fb1-9b09-14be2d21c054)
②The segments looks incorrect
![2wrongvslam](https://github.com/user-attachments/assets/c394e3fc-e17a-4f9e-96e8-1ed1c1b17879)

and one pair of my input rgb and depth is（resolution 1920*1080）：
![renderimg](https://github.com/user-attachments/assets/7348f401-1346-462b-9d52-e92298594001)
![depth](https://github.com/user-attachments/assets/fe1e946a-d532-4141-a9a2-7db665704397)

if you have any idea?

**Comment by luigifreda** [comment]
Hi,
I see now you can show a point cloud.

Can you share your dataset? It isn't easy to understand what is going on.

**Comment by luigifreda** [comment]
As for the depth factor, first, it is read here v
https://github.com/luigifreda/plvs/blob/master/src/Tracking.cc#L1340

and then used here v
https://github.com/luigifreda/plvs/blob/master/src/Tracking.cc#L1812
to convert the depth image to meters.

For instance, see here how the depth image is used to estimate the point depths (in meters)
https://github.com/luigifreda/plvs/blob/master/src/Frame.cc#L2266

In summary, the depth factor decides how you scale your depth measurements in your map. The different pipelines expect the converted depth image return distances in meters.

**Comment by 6NeverGiveup** [comment]
> Hi, I see now you can show a point cloud at least.
> 
> Can you share your dataset? It isn't easy to understand what is going on.

I’m very appreciate for your help
how can I add your contact account, such as qq or wechat or others?

**Comment by luigifreda** [comment]
Could you share a dataset link here?

**Comment by 6NeverGiveup** [comment]
> Could you share a dataset link here?

sure，here is data link
https://drive.google.com/drive/folders/1-UsXH4JamvpYblz5w68mGNjYvhB5u7aM?usp=sharing
some explanations:
1. rgb data comes from multi perspective rendering of the 3D model, depth data calculates the Euclidean distance from pixels to the camera plane, and in previous tests, I found that DepthMapFactor = 40 can produce some information;
2. The scene is an indoor, underground pipeline room.

**Comment by luigifreda** [comment]
Hi @6NeverGiveup ,
Unfortunately, I can't download the shared data. I sent you an access request.

**Comment by 6NeverGiveup** [comment]
> Hi @6NeverGiveup , Unfortunately, I can't download the shared data. I sent you an access request.

sorry, I check my message box, and i didn't get your request msg;
and I tried to diliver data via BaiDu Netdisk as link:
通过百度网盘分享的文件：google_drive.zip
链接：https://pan.baidu.com/s/1ZAPrLc9MzszL8QgtPgAIxQ?pwd=edq8 
提取码：edq8 
--来自百度网盘超级会员V4的分享
please try it, thank u!

**Comment by luigifreda** [comment]
I tried with Badu. However, I am lost in translation :-) 
Could you simply add my email address (see in the README) to the users who are authorized to access your Google Drive link?

**Comment by 6NeverGiveup** [comment]
> I tried with Badu. However, I am lost in translation :-) Could you simply add my email address (see in the README) to the users who are authorized to access your Google Drive link?

Yes, I've passed ur request in google drive and I think now u can touch my data, named "data1206-plvs"
Thank u for ur help! 
sincerely

**Comment by luigifreda** [comment]
Thanks. I was able to download the files. However, I didn't find a file with the camera intrinsics/settings. What are the camera instrinsics you used to generate the images and depths?

**Comment by 6NeverGiveup** [comment]
> Thanks. I was able to download the files. However, I didn't find a file with the camera intrinsics/settings. What are the camera instrinsics you used to generate the images and depths?

The data comes from a 3D rendering model, so those params are very typical, and I made them as follow:
for 1920*1080：
width:1920, height :1080, fx = fy = 1080, cx&cy are half of width&height: cx 960 cy 540
for 640*480:
width:640, height :480, fx = fy = 480, cx&cy are half of width&height: cx 320 cy 240
I don't know if those params obtained from this approach is really correct, but I think so

**Comment by luigifreda** [comment]
Hi,
I’ve been experimenting with your data and had a few questions.

How did you generate the images and depth information? 
If you have control over the rendering process, it should be possible to extract all the necessary details. To achieve meaningful results, we need precise information about the camera intrinsics and the depth factor. Unfortunately, there’s no room for approximation if you want to get a correct map and segmentation.

Some reference formulas 
```
fx = width / (2 * tan(FoV_x / 2))
fy = height / (2 * tan(FoV_y / 2))
cx = width / 2
cy = height / 2

FoV_y = 2 * atan(tan(FoV_x / 2) / aspect_ratio)
FoV_x = 2 * atan(aspect_ratio * tan(FoV_y / 2))
```
Where:
```
- FoV_x = Horizontal field of view (in radians)
- FoV_y = Vertical field of view (in radians)
- aspect_ratio = width / height
```

**Comment by 6NeverGiveup** [comment]
> Hi, I’ve been experimenting with your data and had a few questions.
> 
> How did you generate the images and depth information? If you have control over the rendering process, it should be possible to extract all the necessary details. To achieve meaningful results, we need precise information about the camera intrinsics and the depth factor. Unfortunately, there’s no room for approximation if you want to get a correct map and segmentation.
> 
> Some reference formulas
> 
> ```
> fx = width / (2 * tan(FoV_x / 2))
> fy = height / (2 * tan(FoV_y / 2))
> cx = width / 2
> cy = height / 2
> 
> FoV_y = 2 * atan(tan(FoV_x / 2) / aspect_ratio)
> FoV_x = 2 * atan(aspect_ratio * tan(FoV_y / 2))
> ```
> 
> Where:
> 
> ```
> - FoV_x = Horizontal field of view (in radians)
> - FoV_y = Vertical field of view (in radians)
> - aspect_ratio = width / height
> ```

Thank you for your suggestion! I'll go take a look again.
I didn't really notice these items when obtaining images from the rendering model.
So I set the parameters according to the standard method, which may have caused errors.
Do you have any suggestions for debugging other parameters, I hope you can elaborate on it, I want to achieve a better result
